htslib/vcf.h

```diff
@@ -1230,7 +1230,7 @@ set to one of BCF_ERR* codes and must be checked before calling bcf_write().
     #define bcf_hdr_id2number(hdr,type,int_id)  ((hdr)->id[BCF_DT_ID][int_id].val->info[type]>>12)
     #define bcf_hdr_id2type(hdr,type,int_id)    (uint32_t)((hdr)->id[BCF_DT_ID][int_id].val->info[type]>>4 & 0xf)
     #define bcf_hdr_id2coltype(hdr,type,int_id) (uint32_t)((hdr)->id[BCF_DT_ID][int_id].val->info[type] & 0xf)
-    #define bcf_hdr_idinfo_exists(hdr,type,int_id)  ((int_id)>=0 && bcf_hdr_id2coltype((hdr),(type),(int_id))!=0xf)
+    #define bcf_hdr_idinfo_exists(hdr,type,int_id)  ((int_id)>=0 && (int_id)<(hdr)->n[BCF_DT_ID] && hdr->id[BCF_DT_ID][int_id].val && bcf_hdr_id2coltype((hdr),(type),(int_id))!=0xf)
```

Comment:
You're missing brackets around `hdr` here.  Also `int_id`, but I guess that's less likely to be a problem as it's inside a set of square brackets already.

Although weren't you thinking of changing this to a static inline function?  There could be some benefit given that the parameters get expanded quite a few times now.

Reply:
When we were discussing it I thought it would require more code, then realized it can be easily added to the macro. If that's okay that way, I'll just add the forgotten brackets.